crates/optimism/consensus/src/validation/canyon.rs

```diff
@@ -0,0 +1,42 @@
+//! Canyon consensus rule checks.
+
+use alloy_consensus::BlockHeader;
+use alloy_trie::EMPTY_ROOT_HASH;
+use reth_consensus::ConsensusError;
+use reth_primitives::GotExpected;
+use reth_primitives_traits::BlockBody;
+
+use crate::OpConsensusError;
+
+/// Validate that withdrawals in block body (Shanghai) is always empty in Canyon.
+// todo: link OP docs
```

Comment:
https://specs.optimism.io/protocol/rollup-node-p2p.html#block-validation - stated in block validation rules `[REJECT] if the block is on a topic >= V2 and does not have an empty withdrawals list`. Canyon blocks are broadcast on `V2` topic.